### ISSUE TITLE
Allow disabling topological order for commits

### ISSUE THREAD

**Issue by orhun** [orhun/git-cliff#804]
### Is there an existing issue or pull request for this?

- [X] I have searched the existing issues and pull requests

### Feature description

See #800 for the problem description. To put it briefly, there might be edge cases where the changelog commits aren't ordered correctly when the merge commits are used. 

### Desired solution

Topological order was introduced as default behavior in #415. This behavior should be configurable (e.g. as date order) in the configuration file as follows:

```toml
# sort the commits topologically
topo_order_commits = false
```

We can also rename `topo_order` option to `topo_order_tags` for better alignment between these two options. 

I'm not the best at naming. Maybe @Cyclonit have better ideas since he worked on #541 

### Alternatives considered

None.

### Additional context

See #800

**Comment by orhun** [comment]
This will probably fix #498 and #585

**Comment by alerque** [comment]
I'm guessing this is the issue I'm having with [decasify](https://github.com/alerque/decasify/) whereby if I regenerate the changelog it nukes any record of the v0.6.0 release from orbit and stuffs the only related commit (from a long running breaking change branch) into the v0.5.8 changelog where it does **not** belong.

**Comment by orhun** [comment]
Yeah... It's not super ideal but we should turn my suggestion solution into reality for the next release.

**Comment by lxb1226** [comment]
Is anyone work for this, may be i can try it?

**Comment by orhun** [comment]
Please go ahead @lxb1226 if you are still interested!
